worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go

```diff
@@ -962,7 +973,12 @@ func createSqlUpdateBenthosConfig(
 			out := buildMysqlOutputQueryAndArgs(newResp, tm, tableKey, colSourceMap)
 			output = out
 		}
-		newResp.Columns = output.Columns
+		if newResp.Config.Input.SqlSelect != nil {
+			newResp.Config.Input.SqlSelect.Where = insertConfig.Config.Input.SqlSelect.Where // keep the where clause the same as insert
+			newResp.Columns = output.Columns
+		} else if newResp.Config.Input.PooledSqlRaw != nil {
+			newResp.Config.Input.PooledSqlRaw.Query = insertConfig.Config.Input.PooledSqlRaw.Query // todo: should this be updated to include the column subset?
```

Comment:
@alishakawaguchi - is this okay here? 
For the sql select looks like we provide an updated subset of columns, but the same WHERE statement.
However, for the raw we just have the fully built sql statement, it wont be as easy to hot swap out portions of the query.
Is it imperative that we provide the updated column set here? Trying to understand the reasoning behind it. Maybe we can chat about it offline in office.